### PR TITLE
TNO-2949 Prevent sending report with no subscribers 

### PR DESCRIPTION
**Pull request by kkwangsir**
disable sending button in editor and subscriber 
prevent any sending with no subscribers in reporting service, and set status to failed.

image

### CODE REVIEW

File: app/subscriber/src/features/my-reports/ReportPreview.tsx [hunk 1]
```diff
@@ -194,10 +195,18 @@ export const ReportPreview = ({ report, onFetch, onClose }: IReportPreviewProps)
...
+              title={
+                !hasSubscribers
+                  ? 'There are no subscribers for this report. Add subscribers to enable sending.'
+                  : ''
+              }
+            >
               Send
               <FaPaperPlane />
-            </Button>
+            </Button>{' '}
```
Comment (Fosol) [inline app/subscriber/src/features/my-reports/ReportPreview.tsx:209]: Do you need this space?  Wouldn't it be better to use css to accomplish this?

**Review by Fosol — COMMENTED** [review]
Can you confirm that after a failure due to not having subscribers, that you can then correct it by adding subscribers and then manually sending.

**Comment by kkwangsir** [comment]
> Can you confirm that after a failure due to not having subscribers, that you can then correct it by adding subscribers and then manually sending.
yes. I have tested this case. will tweak the page first.

**Review by Fosol — APPROVED** [review]

**Comment by kkwangsir** [comment]
image

**Pull request merged**